src/io/subhalo_io.cpp

```diff
@@ -307,6 +307,11 @@ void SubhaloSnapshot_t::Save(MpiWorker_t &world)
 
   // Every rank should have executed the writing code exactly once
   assert(writes_done == 1);
+
+  /* Clean up the source of subhaloes from duplicate particles */
+  CleanTracks();
```

Comment:
TODO: move this between the saving of bound subhalo properties and source subhalo properties

Reply:
This will likely require splitting `WriteFile(world.rank(), world.size(), NumSubsAll);` into a `WriteBoundFile` and `WriteSourceFile`